### PR TITLE
Fix MFA without session

### PR DESCRIPTION
**Pull request by gonzaloriestra**
Closes https://github.com/CartoDB/cartodb/issues/14564
Closes https://github.com/CartoDB/cartodb/issues/14566

I couldn't add tests for the redirection because it requires working with Warden and it's hard to stub with Rspec.

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -197,15 +197,16 @@ def check_cors_headers_for_whitelisted_origin
...
     if current_user.nil? || current_user.username != viewed_username
       user = Carto::User.find_by_username(viewed_username)
       render_locked_owner if user.try(:locked?)
     elsif current_user.locked?
       render_locked_user
-    elsif multifactor_authentication_required?
-      render_multifactor_authentication
     end
+
+    render_multifactor_authentication if multifactor_authentication_required?
```
Comment (alrocar) [inline app/controllers/application_controller.rb]: when any of the previous conditions is true, wouldn't this rise a double render error?

for example, what happens when a user is locked but also had multifactor authentication activated?

Comment (alrocar) [reply app/controllers/application_controller.rb]: Just tested and there's the DoubleRenderError for a locked user with MFA.

Other than that the two issues seem to be fixed with this PR 👍

Comment (gonzaloriestra) [reply app/controllers/application_controller.rb]: You are right, good catch!

Although I think there's another bug with the lockout view. I'll create another issue for it.

Comment (gonzaloriestra) [reply app/controllers/application_controller.rb]: https://github.com/CartoDB/cartodb/issues/14607

Comment (alrocar) [reply app/controllers/application_controller.rb]: Is this a side effect of this PR? I'm thinking if we should fix it as part of this PR or not...

Comment (gonzaloriestra) [reply app/controllers/application_controller.rb]: No, it's happening in master

**Comment by alrocar** [comment]
Acceptance 🍏

**Pull request merged**